gnosis/safe-react
# fix: reset safe info when URL changes

## What it solves
Resolves #3327 and #3342

## How this PR fixes it
UPDATE_SAFE wasn't triggering an actual store update because a Safe info payload with just an address in it is equal to the null state. I've added a default value for the owners field (an empty array) to circumvent this.

I've also refactored and simplified a few places related to initial Safe loading.

## How to test it
* Load a safe
* Make network offline
* Switch to another safe
* It should display the new Safe address in the sidebar and keep trying to load the rest of the data
